pkgs/by-name/sk/skypilot/package.nix

```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  pkgs,
+  stdenv,
+  fetchFromGitHub,
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "skypilot";
+  version = "0.6.0";
+
+  src = fetchFromGitHub {
+    owner = "skypilot-org";
+    repo = "skypilot";
+    rev = "v${version}";
+    hash = "sha256-SddXouful2RSp7ijx6YLzfBhBvzN9xKM2dRzivgNflw==";
+  };
+
+  pyproject = true;
+
+  build-system = with python3Packages; [ setuptools ];
+
+  propagatedBuildInputs = with python3Packages; [
+    cachetools
+    click # click >7.0 is required; NixOS 24.05 provides 8.1.7
+    colorama
+    cryptography
+    filelock # filelock >3.6.0 is required; NixOS 24.05 provides 3.13.4
+    jinja2 # jinja2 >3.0 is required; NixOS 24.05 provides 3.1.4
+    jsonschema
+    networkx
+    packaging
+    pandas # pandas >1.3.0 is required; NixOS 24.05 provides 2.2.1
+    pendulum
+    prettytable # PrettyTable >2.0.0 is required; NixOS 24.05 provides 3.9.0
+    psutil
+    python-dotenv
+    pyyaml # pyyaml >3.13 != 5.4 is required; NixOS provides 6.0.1
+    pulp
+    requests
+    rich
+    tabulate
+    # From source (skypilot v0.6.0)
+    # Light weight requirement, can be replaced with "typing" once
+    # we deprecate Python 3.7 (this will take a while).
```

Comment:
Please drop all of those comments. They will be insanely quick out of date and we generally do not care about those precise details unless they have some importance.

Reply:
Thanks @SuperSandro2000 - I have removed them but added a comment indicating that when updating, the `setup.py` file should be checked to ensure package versions are ok. Hope that's ok.